processors/tenantidprocessor/tenantidprocessor.go

```diff
@@ -0,0 +1,68 @@
+package tenantidprocessor
+
+import (
+	"context"
+	"fmt"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"strings"
+
+	"go.opencensus.io/stats/view"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/metadata"
+)
+
+type processor struct {
+	tenantIDHeaderName   string
+	tenantIDAttributeKey string
+	logger               *zap.Logger
+	tenantIDViews        map[string]*view.View
+}
+
+var _ processorhelper.TProcessor = (*processor)(nil)
+
+// ProcessTraces implements processorhelper.TProcessor
+func (p *processor) ProcessTraces(ctx context.Context, traces pdata.Traces) (pdata.Traces, error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		p.logger.Error("Could not extract headers from context", zap.Int("num-spans", traces.SpanCount()))
+		return traces, fmt.Errorf("missing header %s", p.tenantIDHeaderName)
+	}
+
+	tenantIDHeaders := md.Get(p.tenantIDHeaderName)
+	if len(tenantIDHeaders) == 0 {
+		return traces, nil
+	} else if len(tenantIDHeaders) > 1 {
```

Comment:
is there ever a case where this would be valid?  I'm inclined to err on the side of caution and 
 just drop spans which have multiple tenant header values.

Reply:
TBH I don't know, it's probably safer to return an error in this case. Every user should have assigned a single tenant.